File: Sample Code/ObjcSampleCode/Demo/DJISdkDemo/Demo/RemoteController/RCActionsTableViewController.m
//
//  RCActionsTableViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//

#import "RCActionsTableViewController.h"
#import "RCHardwareStateViewController.h"
#import "RCParingViewController.h"
#import "DemoSettingItem.h"
#import "DemoComponentHelper.h"
#import <DJISDK/DJISDK.h>

@interface RCActionsTableViewController ()

@end

@implementation RCActionsTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.sectionNames = [NSMutableArray arrayWithArray:@[@"General"]];
    
    DemoSettingItem* item1 = [DemoSettingItem itemWithName:@"RCHardwareState" andClass:[RCHardwareStateViewController class]];
    DemoSettingItem* item2 = [DemoSettingItem itemWithName:@"RCParing" andClass:[RCParingViewController class]];
    [self.items addObject:@[item1, item2]];

}

-(DJIBaseComponent *)getComponent {
    return [DemoComponentHelper fetchRemoteController]; 
}


@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/RemoteController/RCActionsTableViewController.swift
//
//  RCActionsTableViewController.swift
//  DJISDKSwiftDemo
//
//  Created by DJI on 16/1/6.
//  Copyright © 2016 DJI. All rights reserved.
//

import DJISDK
class RCActionsTableViewController: DemoTableViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.connectedComponent = ConnectedProductManager.sharedInstance.fetchRemoteController()
        self.showComponentVersionSn = true
        
        self.sectionNames = ["General"]
        let item1: DemoSettingItem = DemoSettingItem(name: "RCHardwareState", andClass: RCHardwareStateViewController.self)
        let item2: DemoSettingItem = DemoSettingItem(name: "RCParing", andClass: RCParingViewController.self)
        self.items.append([item1])
        self.items.append([item2])
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
